### PR TITLE
Fix #813, Add Generic Counter API test

### LINKED ISSUE #813: Add functional tests for cFE Generic Counter APIs
**Is your feature request related to a problem? Please describe.**
Need open source functional tests for certifiability

**Describe the solution you'd like**
Add functional tests for cFE Generic Counter APIs
– CFE_ES_RegisterGenCounter - Register a generic counter.
Generated by Doxygen
10 CONTENTS
– CFE_ES_DeleteGenCounter - Delete a generic counter.
– CFE_ES_IncrementGenCounter - Increments the specified generic counter.
– CFE_ES_SetGenCount - Set the specified generic counter.
– CFE_ES_GetGenCount - Get the specified generic counter count.
– CFE_ES_GetGenCounterIDByName - Get the Id associated with a generic counter name.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by jphickey**
**Describe the contribution**
Add tests for the following APIs:

- CFE_ES_RegisterGenCounter
- CFE_ES_CounterID_ToIndex
- CFE_ES_GetGenCounterIDByName
- CFE_ES_GetGenCounterName
- CFE_ES_GetGenCount
- CFE_ES_SetGenCount
- CFE_ES_IncrementGenCounter
- CFE_ES_DeleteGenCounter

Fixes #813

**Testing performed**
Build and run all functional tests, confirm passing

**Expected behavior changes**
Additional APIs are tested

**System(s) tested on**
Ubuntu

**Contributor Info - All information REQUIRED for consideration of pull request**
Joseph Hickey, Vantage Systems, Inc.

### CODE REVIEW

**Comment by jphickey** [comment]
This also depends on nasa/osal#1125, which is why CI is failing right now.  Once the next OSAL merge cycle is complete should be able to re-run CI and confirm success....

**Review by zanzaben — DISMISSED** [review]
Every test should have a UtPrintf("Testing: ..."); at the top stating which functions are being tested.

**Review by skliper — APPROVED** [review]

**Comment by astrogeco** [comment]
[CCB:2021-08-04](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-04-2021) APPROVED with CHANGES
- Add UTprintf

**Comment by jphickey** [comment]
Removing dependency label, as the required change is now in OSAL main branch, everything should be fine to merge normally now.

File: modules/cfe_testcase/src/es_counter_test.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    UtPrintf("Testing: CFE_ES_CounterID_ToIndex");
+
+    /* Confirm CFE_ES_CounterID_ToIndex works (nominal) */
+    Idx = UINT32_MAX;
+    UtAssert_INT32_EQ(CFE_ES_CounterID_ToIndex(TestId, &Idx), CFE_SUCCESS);
+    UtAssert_UINT32_LT(Idx, CFE_PLATFORM_ES_MAX_GEN_COUNTERS);
+
+    /* Confirm proper rejection of bad args in CFE_ES_CounterID_ToIndex */
+    UtAssert_INT32_EQ(CFE_ES_CounterID_ToIndex(CFE_ES_COUNTERID_UNDEFINED, &Idx), CFE_ES_ERR_RESOURCEID_NOT_VALID);
```
Comment (zanzaben) [inline modules/cfe_testcase/src/es_counter_test.c:80]: CFE_ES_CounterID_ToIndex is being tested by https://github.com/nasa/cFE/pull/1716 so you don't need to check it here.

Comment (jphickey) [reply modules/cfe_testcase/src/es_counter_test.c:80]: Yeah, we wrote these tests in parallel.  I don't see a real problem with testing it in two places, as they are slightly different variations.  It doesn't hurt anything.

Comment (zanzaben) [reply modules/cfe_testcase/src/es_counter_test.c:80]: alright just wanted to make sure we were aware.

**Review by zanzaben — APPROVED** [review]

**Pull request merged**